### PR TITLE
Change authentication method to use OIDC identity tokens 

### PR DESCRIPTION
**Pull request by deadlycoconuts**
**What this PR does / why we need it**:
As of present, authentication is performed by the Merlin and Turing SDKs using Google OAuth 2.0 access tokens (generated after authentication with Google's OAuth server) that are added to the headers of requests sent to the Turing API server. These tokens are then used by an internal proxy which will validate the access token and then retrieve the user profile information from the Google API. 

However, in keeping with the move of the Google Identity Services client library from OAuth 2.0 access tokens to Google OpenID Connect (OIDC) identity tokens, we are similarly replacing the use of the former for the latter. These tokens generated would similarly be added to the headers of requests sent to the Turing API server to be validated internally. 

A utils module `caraml_auth` in the package `caraml-auth-google` has been created in the CaraML SDK [repository](https://github.com/caraml-dev/caraml-sdk) to provide this new method of authentication.

This PR serves to update the existing authentication method used by the Turing SDK.

**Which issue(s) this PR fixes**:
The Merlin SDK will import and use the function `get_default_id_token_credentials` provided by the `caraml-auth-google` package:

```python
from caraml_auth.id_token_credentials import get_default_id_token_credentials

credentials = get_default_id_token_credentials(target_audience="sdk.caraml")
credentials.refresh(Request())
```

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: python/sdk/docs/requirements_docs.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ mlflow>=1.2.0,<=1.23.0
 google-cloud-storage>=1.19.0
 urllib3>=1.23
 PyPrind>=2.11.2
-google-auth>=1.11.0,<2.0dev
+google-auth>=2.16.2
```
Comment (krithika369) [inline python/sdk/docs/requirements_docs.txt:20]: Same comment as https://github.com/caraml-dev/turing/pull/324/files#r1134052734

(Also about the `setup.py` file below.)

**Review by krithika369 — APPROVED** [review]
Left a small comment. The rest LGTM. Thanks @deadlycoconuts !

File: python/sdk/docs/requirements_docs.txt (hunk 1 above)
Comment (deadlycoconuts) [reply python/sdk/docs/requirements_docs.txt]: Yep, I think it's redundant too. Just removed it! (In the `setup.py` file too)

**Comment by deadlycoconuts** [comment]
Thanks a lot for the quick review! Will be merging this once the pipeline succeeds!

**Pull request merged**